chore: disable warning about pre C99 code

sudden6 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/toktok/c-toxcore/2318)
<!-- Reviewable:end -->

iphydf committed:
cleanup: Remove check_recursion, since tokstyle checks this now.

And tokstyle is better at it, covering not just direct recursion, but
also (most possible instances of) recursion through function pointers.

dependabot[bot] committed:
chore(deps): bump third_party/cmp from `4c3737d` to `e836703`

Bumps [third_party/cmp](https://github.com/camgunz/cmp) from `4c3737d` to `e836703`.
- [Release notes](https://github.com/camgunz/cmp/releases)
- [Commits](https://github.com/camgunz/cmp/compare/4c3737d1a172ee74d2b35b5fd6865d1fe9dac7af...e836703291392aba9db92b46fb47929521fac71f)

---
updated-dependencies:
- dependency-name: third_party/cmp
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TokTok/c-toxcore?pullRequest=2318) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/TokTok/c-toxcore?pullRequest=2318) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: sudden6<br/>:x: dependabot[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/TokTok/c-toxcore?pullRequest=2318) it.</sub>

sudden6 committed:
test: fix wrong log callback types

sudden6 committed:
refactor: Simplify locking logic in DHT

This fixes an error reported by cppcheck about overflowing the callback
array. The locking logic probably was broken, so I replaced it with
something simpler and more robust.

sudden6 committed:
test: remove outdated cppcheck option

sudden6 committed:
test: disable cppcheck rules that trigger false positives

sudden6 committed:
chore: update bootstrapd hash

JFreegman reviewed (approved):
> Reviewed 1 of 1 files at r1, 4 of 4 files at r2, 8 of 9 files at r3, 1 of 2 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/2318)* status: :shipit: complete! 2 of 1 approvals obtained



<!-- Sent from Reviewable.io -->
